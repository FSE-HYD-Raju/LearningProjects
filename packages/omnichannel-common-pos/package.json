{
	"name": "omnichannel-common-pos",
	"version": "111.2.0",
	"description": "Common actions, stores and components for omnichannel",
	"main": "lib/index",
	"types": "lib/index.d.ts",
	"oc:es6": "src/index",
	"scripts": {
		"test": "../../testSinglePackage.sh $(pwd)",
		"test-watch": "jest --watch",
		"lint": "eslint tests src --max-warnings=0 ",
		"test-ci": "jest",
		"lint-fix": "eslint src tests --fix",
		"flow": "flow; test $? -eq 0 -o $? -eq 2",
		"clean": "rm -rf lib",
		"build-ts": "tsc --pretty",
		"copy-assets": "copyfiles --up 1 'src/**/*.@(scss|jpg|jpeg|gif|png|svg)' ./lib",
		"build": "cross-env NODE_ENV=production BABEL_ENV=production BABEL_DISABLE_CACHE=1 babel -s inline -d lib/ src/ && npm run build-ts && npm run copy-assets && rimraf -rf messages",
		"watch": "cross-env NODE_ENV=production BABEL_ENV=production BABEL_DISABLE_CACHE=1 babel -w -s inline -d lib/ src/ && rimraf -rf messages",
		"prepublish": "npm run clean && npm run build"
	},
	"jest": {
		"testRegex": "(/tests/.*)\\.test\\.js$",
		"coverageDirectory": "jestcoverage",
		"coverageReporters": [
			"lcov",
			"html"
		],
		"setupFiles": [
			"../../jest-setup.js"
		]
	},
	"author": "",
	"license": "ISC",
	"publishConfig": {
		"registry": "https://artifactory.qvantel.net/artifactory/api/npm/npm-local"
	},
	"dependencies": {
		"alt": "0.18.6",
		"alt-utils": "2.0.0",
		"autosuggest-highlight": "2.1.1",
		"axios": "0.18.0",
		"babel-cli": "6.26.0",
		"babel-core": "6.26.0",
		"babel-eslint": "7.2.3",
		"babel-jest": "23.6.0",
		"babel-loader": "7.1.3",
		"babel-plugin-add-module-exports": "0.2.1",
		"babel-plugin-module-resolver": "3.1.0",
		"babel-plugin-react-intl": "2.4.0",
		"babel-plugin-react-transform": "2.0.2",
		"babel-plugin-transform-class-properties": "6.24.1",
		"babel-plugin-transform-decorators-legacy": "1.3.4",
		"babel-plugin-transform-es2015-classes": "6.24.1",
		"babel-plugin-transform-proto-to-assign": "6.26.0",
		"babel-plugin-transform-react-constant-elements": "6.23.0",
		"babel-plugin-transform-react-inline-elements": "6.22.0",
		"babel-plugin-transform-react-pure-class-to-function": "1.0.1",
		"babel-plugin-transform-react-remove-prop-types": "0.4.20",
		"babel-plugin-transform-runtime": "6.23.0",
		"babel-plugin-typecheck": "3.9.0",
		"babel-polyfill": "6.26.0",
		"babel-preset-es2015": "6.24.1",
		"babel-preset-es2015-loose": "7.0.0",
		"babel-preset-react": "6.24.1",
		"babel-preset-stage-0": "6.24.1",
		"babel-register": "6.26.0",
		"babel-runtime": "6.26.0",
		"bootstrap-sass": "3.3.7",
		"byte-size": "4.0.4",
		"classnames": "2.2.6",
		"connected-react-router": "4.3.0",
		"copyfiles": "2.0.0",
		"country-data": "0.0.31",
		"cross-env": "5.1.4",
		"crypto-js": "3.1.9-1",
		"css-ns": "1.2.1",
		"debug": "4.1.0",
		"diff": "3.4.0",
		"history": "4.7.2",
		"jwt-decode": "2.2.0",
		"langs": "2.0.0",
		"lodash": "4.17.11",
		"moment": "2.21.0",
		"moment-duration-format": "2.2.2",
		"object-assign": "4.1.1",
		"omnichannel-flow-pos": "111.2.0",
		"path": "0.12.7",
		"path-to-regexp": "2.3.0",
		"prop-types": "15.6.1",
		"ramda": "0.24.1",
		"rc-progress": "2.2.5",
		"rc-slider": "8.6.1",
		"react": "16.10.2",
		"react-autosuggest": "9.3.4",
		"react-bootstrap": "0.32.1",
		"react-collapse": "4.0.3",
		"react-confirm": "0.1.14",
		"react-dates": "20.1.0",
		"react-dom": "16.10.2",
		"react-dropzone": "3.13.4",
		"react-flatpickr": "3.6.4",
		"react-formal": "0.25.4",
		"react-formal-inputs": "2.1.1",
		"react-helmet": "5.2.0",
		"react-highlight-words": "0.11.0",
		"react-input-range": "1.3.0",
		"react-intl": "2.7.0",
		"react-motion": "0.5.2",
		"react-motion-ui-pack": "0.10.3",
		"react-onclickoutside": "6.7.1",
		"react-redux": "5.0.7",
		"react-router": "4.2.0",
		"react-router-dom": "4.2.2",
		"react-select": "1.2.1",
		"react-toastr": "3.0.0",
		"react-widgets": "3.5.0",
		"recompose": "0.30.0",
		"redux": "4.0.0",
		"redux-saga": "0.16.0",
		"redux-saga-tester": "1.0.379",
		"shallowequal": "0.2.2",
		"twix": "1.2.1",
		"universal-cookie": "2.1.2",
		"uuid": "3.2.1",
		"weasel-styles": "2.3.1",
		"yup": "0.21.3"
	},
	"devDependencies": {
		"@types/alt": "0.16.35",
		"@types/classnames": "2.2.6",
		"@types/jest": "23.3.9",
		"@types/lodash": "4.14.118",
		"@types/moment-duration-format": "2.2.2",
		"@types/node": "8.0.0",
		"@types/rc-slider": "8.2.3",
		"@types/react": "16.7.18",
		"@types/react-bootstrap": "0.32.14",
		"@types/react-collapse": "4.0.2",
		"@types/react-confirm": "0.1.2",
		"@types/react-dates": "17.1.5",
		"@types/react-dom": "16.0.5",
		"@types/react-flatpickr": "3.7.0",
		"@types/react-helmet": "5.0.7",
		"@types/react-highlight-words": "0.12.0",
		"@types/react-intl": "2.3.9",
		"@types/react-motion": "0.0.27",
		"@types/react-motion-ui-pack": "0.10.1",
		"@types/react-onclickoutside": "6.0.4",
		"@types/react-redux": "6.0.0",
		"@types/react-router": "4.0.25",
		"@types/react-select": "1.2.0",
		"@types/react-toastr": "3.0.3",
		"@types/recompose": "0.26.4",
		"@types/redux-actions": "2.3.0",
		"@types/redux-devtools": "3.0.44",
		"@types/redux-devtools-dock-monitor": "1.1.32",
		"@types/redux-devtools-log-monitor": "1.0.33",
		"@types/redux-logger": "3.0.6",
		"@types/seamless-immutable": "7.1.2",
		"@types/universal-cookie": "2.2.0",
		"@types/webpack": "4.4.0",
		"@types/webpack-env": "1.13.6",
		"@types/yup": "0.26.0",
		"babel-plugin-__coverage__": "11.0.0",
		"babel-plugin-transform-flow-strip-types": "6.22.0",
		"babel-plugin-transform-runtime": "6.23.0",
		"enzyme": "3.7.0",
		"enzyme-adapter-react-16": "1.10.0",
		"enzyme-to-json": "3.3.3",
		"eslint": "4.10.0",
		"eslint-config-omnichannel": "2.2.4",
		"eslint-config-prettier": "1.7.0",
		"eslint-config-react-app": "2.1.0",
		"eslint-loader": "1.9.0",
		"eslint-plugin-babel": "4.1.2",
		"eslint-plugin-flowtype": "2.39.1",
		"eslint-plugin-import": "2.8.0",
		"eslint-plugin-jsx-a11y": "5.1.1",
		"eslint-plugin-prettier": "2.6.0",
		"eslint-plugin-react": "7.4.0",
		"eslint-plugin-react-intl": "1.1.1",
		"flow-bin": "0.70.0",
		"jest": "23.6.0",
		"jest-junit": "3.6.0",
		"lint-staged": "4.3.0",
		"moxios": "0.4.0",
		"pre-commit": "1.2.2",
		"prettier": "1.6.1",
		"react-router-enzyme-context": "1.2.0",
		"react-test-renderer": "16.7.0",
		"redux-devtools": "3.4.1",
		"redux-devtools-dock-monitor": "1.1.3",
		"redux-devtools-log-monitor": "1.4.0",
		"redux-logger": "3.0.6",
		"rimraf": "2.6.2",
		"sinon": "4.3.0",
		"ts-jest": "23.10.5",
		"ts-loader": "5.3.1",
		"typescript": "3.1.6",
		"webpack": "4.26.1"
	},
	"commitInfo": "omni-ui-common-pos|184f176|1571059113|Venkata_Gangadhar_Kodela|develop"
}
