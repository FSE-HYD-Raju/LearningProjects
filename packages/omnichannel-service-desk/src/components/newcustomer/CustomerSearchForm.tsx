import * as React from "react";
import { Component } from "react";
import { get, includes, isEmpty } from "lodash";
import { withSchema } from "omnichannel-common-pos";
import CustomerDataFormContainer from "../CustomerDataFormContainer";
import { Schema } from "yup";

interface CustomerSearchFormStateProps {
	schema?: Schema<any> & { fields: Array<any> };
	displayOptions: {
		organizationIdentification: object;
		identification: Array<any>;
		customerDataForm: any;
	};
	matchForIdTypeAndNumber: Array<any>;
	customerIdValidationStatus: boolean;
	customerIdentificationValidationEnabled: any;
	additionalFields: any;
	customerToCreate: any;
}

interface CustomerSearchFormActionProps {
	actions: {
		searchCustomers: (firstName: string, lastName: string) => void;
		saveSearchTerms: (model: any) => void;
		validateIdentification: (idType: string, idNumber: number, familyName: string) => void;
		searchCustomerWithSingleTerm: (idNumber: string, idType: string) => void;
	};
}

interface CustomerSearchFormState {
	model: any;
	validationErrors: Array<any>;
}

type CustomerSearchFormProps = CustomerSearchFormStateProps & CustomerSearchFormActionProps;

class CustomerSearchForm extends Component<CustomerSearchFormProps, CustomerSearchFormState> {
	static displayName = "CustomerSearchForm";

	constructor(props: CustomerSearchFormProps) {
		super(props);

		this.state = {
			model: props.customerToCreate || {},
			validationErrors: []
		};
	}

	doSearch = (model: any) => {
		const { schema,
				displayOptions,
				matchForIdTypeAndNumber,
				customerIdValidationStatus,
				customerIdentificationValidationEnabled,
				actions
			  } = this.props;
		setTimeout(() => {
			if (schema) {
				schema.isValid(model).then((valid: any) => {
					if (valid) {
						const firstName = get(model, "firstName");
						const lastName = get(model, "lastName");

						const identificationType = get(model, "identificationType");
						const identificationNumber = get(model, "identificationId");

						const identifications = displayOptions.identification || [];
						const matchingIdentificationType = identifications.find((identificationTypeObject: any) => {
							const match = includes(identificationTypeObject.localisation, identificationType);
							return match;
						});
						const identificationTypeBackendValue = matchingIdentificationType.backendValue;
						actions.searchCustomers(firstName, lastName);
						actions.searchCustomerWithSingleTerm(identificationNumber, identificationTypeBackendValue)
						if (isEmpty(matchForIdTypeAndNumber) &&
							!customerIdValidationStatus &&
							customerIdentificationValidationEnabled.enabled &&
							(!!customerIdentificationValidationEnabled.idTypes.find(
								(idType: string) => idType === matchingIdentificationType.backendValue
							))
						) {
							actions.validateIdentification(identificationTypeBackendValue, identificationNumber, lastName);
						}
					}
				});
			}
		}, 500);
	};

	handleBlur = (partialModel: any) =>  {
		this.setState({
				model: {
					...this.state.model,
					...partialModel
				},
			}, () => {
				this.props.actions.saveSearchTerms(this.state.model);
			}
		);
	};

	render() {
		return (
			<div className="new-customer-search">
				<CustomerDataFormContainer
					schema={this.props.schema}
					validationErrors={this.state.validationErrors}
					onSubmit={this.doSearch}
					onBlur={this.handleBlur}
					saveSearchTerms={this.props.actions.saveSearchTerms}
					additionalFields={this.props.additionalFields}
				/>
			</div>
		);
	}
}

export default withSchema(["customerDataForm"])(CustomerSearchForm);

export {
	CustomerSearchFormStateProps,
	CustomerSearchFormActionProps,
	CustomerSearchFormProps,
	CustomerSearchFormState,
};
