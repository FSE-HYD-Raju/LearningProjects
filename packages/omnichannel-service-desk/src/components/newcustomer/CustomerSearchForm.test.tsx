import * as React from "react";
import { TestUtils, shallowWithContext, mountWithContext, SimpleDataMock } from "omnichannel-common-pos";
import CustomerSearchForm, { CustomerSearchFormProps } from "./CustomerSearchForm";

describe("CustomerSearchForm", () => {
	const customCustomerDataForm = {
		$schema: "http://json-schema.org/draft-04/schema#",
		title: "customerDataForm",
		type: "object",
		properties: {
			firstName: {
				type: "string",
				required: true,
				editable: true
			},
			lastName: {
				type: "string",
				required: true,
				editable: true
			},
			email: {
				type: "string",
				required: true,
				editable: true,
				validation: {
					email: true
				}
			},
			mobileNumber: {
				type: "string",
				required: false,
				editable: true
			},
			fixedLineNumber: {
				type: "string",
				required: false,
				editable: true
			},
			birthDay: {
				type: "date",
				required: false,	
				editable: false,
				nullable: true,
				format: ""
			},
			gender: {
				type: "string",
				required: true,
				editable: true
			},
			passportNumber: {
				type: "string",
				required: true,
				editable: true
			},
			passportExpiryDate: {
				type: "string",
				required: true,
				editable: true
			},
			street: {
				type: "string",
				required: false
			},
			coAddress: {
				type: "string",
				required: false
			},
			postalCode: {
				type: "string",
				required: false
			},
			city: {
				type: "string",
				required: false
			},
			country: {
				type: "string",
				required: false
			}
		}
	};

	const context = SimpleDataMock.getConsulContextMock();
	context.store = TestUtils.mockReduxStore({
		schema: {
			schemas: {
				customerDataForm: customCustomerDataForm
			},
		},
		consul: {
			customerCreationMandatoryFields: ["firstname", "lastname", "gender"],
			displayOptions: {
				identification: [],
				organizationIdentification: {},
				customerDataForm: {},
			},
		},
		feature: {},
		cms: {},
	});

	const minProps: CustomerSearchFormProps = {
		displayOptions: {
			identification: [],
			organizationIdentification: {},
			customerDataForm: {},
		},
		matchForIdTypeAndNumber: [],
		customerIdValidationStatus: false,
		customerIdentificationValidationEnabled: false,
		additionalFields: [],
		customerToCreate: {},
		actions: {
			createCustomer: jest.fn(),
			searchCustomers: jest.fn(),
			saveSearchTerms: jest.fn(),
			validateIdentification: jest.fn(),
			searchCustomerWithSingleTerm: jest.fn()
		}
	};

	it("succeeds at shallow mount with min props", () => {
		const wrapper = shallowWithContext(<CustomerSearchForm {...minProps} />, { context });
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount with minimum props", () => {
		const wrapper = mountWithContext(<CustomerSearchForm {...minProps} />, { context });

		const searchForm = wrapper.find(".new-customer-search");
		expect(searchForm.length).toEqual(1);

		const customerDataForm = searchForm.find(".CustomerDataForm");
		expect(customerDataForm.hostNodes()).toHaveLength(1);
	});

	/* CustomerActions.saveSearchTerms() should be called with object that contains the value in email field
	 * data should be found from CustomerStore, but that's not in scope of this test.
	 */
	describe("RUBT-56116", () => {
		it("CustomerActions.saveSearchTerms() should be called at onBlur event with a key-value pair from email field merged with the rest of data", done => {
			let partialModel: any;
			const newProps = {
				...minProps,
				customerToCreate: {
					firstName: "John",
					lastName: "Smith",
					gender: "male",
					email: "",
					mobileNumber: "1234",
					fixedLineNumber: "2345",
					street: "1 Test St.",
					postalCode: "12340",
					city: "Test City",
					country: "Testia"
				},
				customerIdValidationStatus: false,
				customerIdValidationError: "",
				validationServiceDown: false,
				actions: {
					...minProps.actions,
					saveSearchTerms: (model: any) => {
						expect(Object.keys(model)).toContain(partialModel.name);
						expect(model[partialModel.name]).toEqual(partialModel.value);
						done();
					}
				}
			};

			const wrapper = mountWithContext(<CustomerSearchForm {...newProps}/>, { context });

			const iEmail = wrapper.find("#inputEmailIntoCustomerDataForm").hostNodes();

			partialModel = {
				name: iEmail.prop("name"),
				value: "john.smith@qvantel.com"
			};

			iEmail.simulate("blur", { target: partialModel });
			wrapper.update();
		});
	});
});
