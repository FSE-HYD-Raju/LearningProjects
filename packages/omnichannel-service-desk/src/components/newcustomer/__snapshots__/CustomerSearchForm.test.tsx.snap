// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerSearchForm succeeds at shallow mount with min props 1`] = `
<CustomerSearchForm
  actions={
    Object {
      "createCustomer": [MockFunction],
      "saveSearchTerms": [MockFunction],
      "searchCustomerWithSingleTerm": [MockFunction],
      "searchCustomers": [MockFunction],
      "validateIdentification": [MockFunction],
    }
  }
  additionalFields={Array []}
  customerIdValidationStatus={false}
  customerIdentificationValidationEnabled={false}
  customerToCreate={Object {}}
  displayOptions={
    Object {
      "customerDataForm": Object {},
      "identification": Array [],
      "organizationIdentification": Object {},
    }
  }
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  matchForIdTypeAndNumber={Array []}
  schema={
    ObjectSchema {
      "_blacklist": BadSet {
        "_map": Object {},
        "_refs": Object {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [
        "undefined-undefined",
      ],
      "_exclusive": Object {},
      "_label": "customerDataForm",
      "_mutate": false,
      "_nodes": Array [
        "country",
        "city",
        "postalCode",
        "coAddress",
        "street",
        "passportExpiryDate",
        "passportNumber",
        "gender",
        "birthDay",
        "fixedLineNumber",
        "mobileNumber",
        "email",
        "lastName",
        "firstName",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": BadSet {
        "_map": Object {},
        "_refs": Object {},
      },
      "fields": Object {
        "birthDay": DateSchema {
          "_blacklist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {},
          "_label": "birthDay",
          "_mutate": false,
          "_nullable": true,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "date",
          "_typeError": [Function],
          "_whitelist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
        },
        "city": StringSchema {
          "_blacklist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {},
          "_label": "city",
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
        },
        "coAddress": StringSchema {
          "_blacklist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {},
          "_label": "coAddress",
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
        },
        "country": StringSchema {
          "_blacklist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {},
          "_label": "country",
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
        },
        "email": StringSchema {
          "_blacklist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": false,
            "trim": false,
            "undefined": false,
          },
          "_label": "email",
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
            [Function],
          ],
        },
        "firstName": StringSchema {
          "_blacklist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": false,
            "trim": false,
          },
          "_label": "firstName",
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
            [Function],
          ],
        },
        "fixedLineNumber": StringSchema {
          "_blacklist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {},
          "_label": "fixedLineNumber",
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
        },
        "gender": StringSchema {
          "_blacklist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": false,
            "trim": false,
          },
          "_label": "gender",
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
            [Function],
          ],
        },
        "lastName": StringSchema {
          "_blacklist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": false,
            "trim": false,
          },
          "_label": "lastName",
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
            [Function],
          ],
        },
        "mobileNumber": StringSchema {
          "_blacklist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {},
          "_label": "mobileNumber",
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
        },
        "passportExpiryDate": StringSchema {
          "_blacklist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": false,
            "trim": false,
          },
          "_label": "passportExpiryDate",
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
            [Function],
          ],
        },
        "passportNumber": StringSchema {
          "_blacklist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": false,
            "trim": false,
          },
          "_label": "passportNumber",
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
            [Function],
          ],
        },
        "postalCode": StringSchema {
          "_blacklist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {},
          "_label": "postalCode",
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
        },
        "street": StringSchema {
          "_blacklist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {},
          "_label": "street",
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": BadSet {
            "_map": Object {},
            "_refs": Object {},
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
/>
`;
